app/controllers/nsm/steps/start_page_controller.rb

```diff
@@ -15,6 +15,16 @@ def show
         @application = current_application
         render 'laa_multi_step_forms/task_list/show', locals: { header: -> { decision_step_header }, app_type: 'claim' }
       end
+
+      # we remove the check_answers entry from the list to reset the return
+      # location after edits once the task list has been displayed
+      def append_navigation_stack
+        current_application.navigation_stack -= [
```

Comment:
This feels a leetle bit like a abuse of the navigation stack. Essentially we're using it to set a flag saying either "check answers page" or "start page". Doing it by adding and removing from the navigation stack is maybe less tidy than, say, stashing something in the session, or in a standalone database field?

Reply:
I have left this at the moment over adding a separate DB field.... 

my thoughts are the navigation stack contains the locations you have visited and are still allowed to visit.. (used in the `in_progress` check in task list
So visiting the task list removes the `permission` on check_answers... so this maybe what it was intended for

Reply:
Yeah all right